aeron-client/src/main/c/concurrent/aeron_atomic64_c11.h

```diff
@@ -54,14 +60,14 @@ while (false) \
 #define AERON_GET_AND_ADD_INT64(original, dst, value) \
 do \
 { \
-    original = atomic_fetch_add((_Atomic(int64_t) *)&dst, value); \
+    original = atomic_fetch_add((_Atomic(int64_t) *)&dst, (int64_t)value); \
```

Comment:
Why add a cast as this can hide a real type issue?

Reply:
because in function aeron_term_appender_get_and_add_raw_tail, value type is size_t, this will not compile 
for std::atomic<int64_t> in cpp context.
This is a quick fix.
We could also change all occurrences of aeron_term_appender_get_and_add_raw_tail.

```cpp
 // in c/concurrent/aeron_term_appender.h
 inline int64_t aeron_term_appender_get_and_add_raw_tail(volatile int64_t *addr, size_t aligned_length)
 {
     int64_t result;
     AERON_GET_AND_ADD_INT64(result, *addr, aligned_length);
     return result;
 }
```

Reply:
This is a more appropriate fix. https://github.com/real-logic/aeron/commit/2b923e7e0a3f066e93ab14f2cdc2761340086735

Reply:
please review the new commits below.